packages/fx-core/src/common/local/telemetry.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { FxError, TelemetryReporter, UserError } from "@microsoft/teamsfx-api";
+import { Component, TelemetryErrorType, TelemetryProperty, TelemetrySuccess } from "../telemetry";
+
+export enum TelemetryEvent {
+  DetectPortStart = "detect-port-start",
+  DetectPort = "detect-port",
+}
+
+export function sendTelemetryEvent(
```

Comment:
Why not directly use `sendTelemetryEvent` from "../telemetry"?

Reply:
../telemetry use the static instance of TelemetryReporter, https://github.com/OfficeDev/TeamsFx/blob/6c049f00d86ca5b6fe4656ef9d64611642cec47e/packages/fx-core/src/common/telemetry.ts#L7
which is init in `FxCore`.
https://github.com/OfficeDev/TeamsFx/blob/6c049f00d86ca5b6fe4656ef9d64611642cec47e/packages/fx-core/src/core/index.ts#L186

`LocalEnvManager` is directly called from vsc / cli. So here an independent send telemetry method is implemented

Reply:
I think they are the same one. `LocalEnvManager` is always called after core is activated, so telemetry instance from core is the same one passed from vsc/cli.

It seems logProvider and ui are also duplicated from core, one further improvement could be using TOOLS from core and make `LocalEnvManager` static.

Reply:
Got it, changed to use `sendTelemetryEvent` in `../telemetry`